text/0000-pack-buildpack-new-alternatives.md

```diff
@@ -0,0 +1,210 @@
+# Meta
+[meta]: #meta
+- Name: Pack Buildpack New to Support Alternatives
+- Start Date: 2022-03-10
+- Author(s): aidandelaney
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+The `pack buildpack new` subcommand creates a new buildpack based on a shell-script template.  This proposal extends `pack buildpack new` with a `--libcnb` option and a `--template URL` option to allow alternatives to the shell-script template.  The `--libcnb` option creates a new buildpack written in golang using `libcnb`.  The `--template URL` option allows buildpack creation from an arbitrary 3rd party URL.
```

Comment:
I would prefer if we just deprecate `pack buildpack new` (and leave it as it is in the next version of pack with a warning and remove it in a couple of versions) and point them to `pack buildpack create` which is interactive and doesn't take any of the existing CLI flags that `pack buildpack new` does (api, stack etc.). This would allow the template to decide which prompts/flags it takes as inputs. We can have certain trusted/recommended templates similar to trusted builders and users may select one of those as `default` if they wish to.

Reply:
I'm going down that route now.  I've updated the RFC to reflect this.